packages/vendure-scripts/src/assign-these-prodcuts-to-channel.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  AssetService,
+  ChannelService,
+  FacetService,
+  ID,
+  Injector,
+  Product,
+  ProductVariantService,
+  RequestContext,
+  SearchService,
+  TransactionalConnection,
+} from '@vendure/core';
+import { getSuperadminContextInChannel } from '../../test/src/admin-utils';
+
+export async function assignTheseProductsToChannel(
+  targetChannelId: ID,
+  injector: Injector,
+  products: Product[],
+  ctx: RequestContext
+): Promise<void> {
+  // start the transaction
+  const conn = injector.get(TransactionalConnection);
+  await conn.startTransaction(ctx);
+  // assign them to target channel
+  const productVariantService = injector.get(ProductVariantService);
+  const assetService = injector.get(AssetService);
+  const facetService = injector.get(FacetService);
+  const channelService = injector.get(ChannelService);
+  const searchService = injector.get(SearchService);
+  const facetIds = products
+    .map((product) =>
+      (product.facetValues ?? []).map((facetValue) => facetValue.facet.id)
+    )
+    .flat();
+  const assetIds = products
+    .map((product) => (product.assets ?? []).map((asset) => asset.id))
+    .flat();
+  const variantIds = products
+    .map((product) => (product.variants ?? []).map((variant) => variant.id))
+    .flat();
+  const targetChannel = await channelService.findOne(ctx, targetChannelId);
+  const superadminContextInTargetChannel = await getSuperadminContextInChannel(
```

Comment:
Can't we use the request context that the consumer gives us, instead of creating our own super admin credentials?

Reply:
I think the configured SearchPlugin uses the `ctx.channelId` when reindexing. See [this](https://github.com/vendure-ecommerce/vendure/blob/0b1dcb7b03ca127ac8e63540d61d13fbcc02ff9f/packages/core/src/plugin/default-search-plugin/indexer/indexer.controller.ts#L332) for the DefaultSearchPlugin.

Reply:
Goood find :ok_hand: . Can we make it so that we only pass a custom Ctx to the `reindex()` call, and use the given ctx for the other calls? The reason for this is that this script **might** trigger customer project in the consumer code like eventlisteners. That custom code might give unexpected results with a Ctx constructed by us.

Reply:
I think that is the case in the current implementation @martijnvdbrug